File: CS/Form1.cs
using DevExpress.XtraEditors;
using System;
using System.Data;
using System.Linq;
using System.Windows.Forms;

namespace NoFocusedRow
{
    public partial class Form1 : XtraForm
    {
        public Form1() {
            InitializeComponent();
        }
        protected override void OnShown(EventArgs e) {
            base.OnShown(e);
            gridView1.FocusInvalidRow();
        }
        private void Form1_Load(object sender, System.EventArgs e) {
            gridControl1.DataSource = GetData();
        }

        DataTable GetData() {
            DataTable table = new DataTable();
            table.Columns.Add("ID", typeof(int));
            table.Columns.Add("ItemName");
            for (int i = 1; i <= 5; i++)
                table.Rows.Add(new object[] { i, "Item " + i.ToString() });
            table.AcceptChanges();
            return table;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Data
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.XtraEditors

Namespace NoFocusedRow
    Partial Public Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Protected Overrides Sub OnShown(e As EventArgs)
            MyBase.OnShown(e)
            gridView1.FocusInvalidRow()
        End Sub
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
            gridControl1.DataSource = GetData()
        End Sub

        Private Function GetData() As DataTable
            Dim table As New DataTable()
            table.Columns.Add("ID", GetType(Integer))
            table.Columns.Add("ItemName")
            For i As Integer = 1 To 5
                table.Rows.Add(New Object() {i, "Item " & i.ToString()})
            Next i
            table.AcceptChanges()
            Return table
        End Function
    End Class
End Namespace
